### PR TITLE
Procurve hardware description cleanup

### PR DESCRIPTION
**Pull request by murrant**
Include part number and other details where available
Covers 100% of known sysDescr
Adds Aruba Instant On parsing
adds string replacements for OS fields

Could use hardware: ENTITY-MIB::entPhysicalDescr.1. It is not supported on all devices and strings contain a lot of fluff and some are just "HP", I leave that for another PR

Please give a short description what your pull request is for

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [ ] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: tests/data/procurve.json [hunk 1]
```diff
@@ -549,7 +549,7 @@
                     "sysDescr": "HP Stack, revision KA.15.03.3004, ROM KA.15.05 (/sw/code/build/tam(KA_15_03))",
                     "sysContact": "<private>",
                     "version": "KA.15.03.3004",
-                    "hardware": "",
+                    "hardware": "Stack",
```
Comment (Jellyfrog) [inline tests/data/procurve.json:552]: This seems wrong

File: tests/data/procurve_131.json [hunk 2]
```diff
@@ -3,12 +3,12 @@
...
             "devices": [
                 {
-                    "sysName": "",
+                    "sysName": null,
                     "sysObjectID": ".1.3.6.1.4.1.11.2.3.7.11.131",
                     "sysDescr": "HP J9625A 2620-24-PoEP Switch, revision RA.16.02.0012, ROM RA.15.13 (/ws/swbuildm/rel_spokane_qaoff/code/build/xform(swbuildm_rel_spokane_qaoff_rel_spokane))",
                     "sysContact": null,
                     "version": "RA.16.02.0012",
-                    "hardware": "2620-24-PoEP",
+                    "hardware": "J9625A 2620-24-PoEP",
```
Comment (Jellyfrog) [inline tests/data/procurve_131.json:11]: I'm thinking we might want to filter out the JxxxxA part, pattern seems to be `J[A-Z0-9]{4}A` or maybe `J[A-Z0-9][0-9]{3}A`

File: tests/data/procurve.json (hunk 1 above)
Comment (murrant) [reply tests/data/procurve.json:552]: I thought it was more right than "".  This is also something that could be fixed by `ENTITY-MIB::entPhysicalDescr.1` maybe.

File: tests/data/procurve_131.json (hunk 2 above)
Comment (murrant) [reply tests/data/procurve_131.json:11]: I specifically included the part number.  I find it more useful than the model name.

Comment (mpikzink) [reply tests/data/procurve_131.json:11]: I agree to @murrant. Would it make sense to put this information in the (actually empty) features tag?

Comment (murrant) [reply tests/data/procurve_131.json:11]: Not straight forward to do. You have this format: 516733-B21, some end with B, or C, sometimes it is after the model and sometimes before, and sometimes there is no model and only the part number.

Comment (mpikzink) [reply tests/data/procurve_131.json:11]: That's a point...

Comment (murrant) [reply tests/data/procurve_131.json:11]: @mpikzink What I mean is, you can't split out the part number without doubling or more the number of regexes required.

**Review by Jellyfrog — APPROVED** [review]

**Pull request merged**